GitDiffMargin/Git/GitCommands.cs

```diff
@@ -282,5 +325,90 @@ static bool HasPreamble(string file, Encoding encoding)
 
             return true;
         }
+
+        private string AdjustPath(string fullPath)
+        {
+            // Right now the only extension is for *.csproj and *.vbproj, which are opened from a temporary location
+            switch (Path.GetExtension(fullPath).ToLowerInvariant())

```

Comment:
A better approach is probably to first see if it ends in just `proj`, and then do the comparison with project file names.